src/plone/restapi/tests/test_blocks_deserializer.py

```diff
@@ -308,3 +308,82 @@ def test_deserialize_subblocks_transformers(self):
             "html"
         ]
         self.assertEqual(block, u"<div>This stays</div>")
+
+    def test_slate_internal_link_deserializer(self):
+        """test_internal_link_deserializer."""
+        blocks = {
+            "2caef9e6-93ff-4edf-896f-8c16654a9923": {
+                "@type": "slate",
+                "plaintext": "this is a slate link inside some text",
+                "value": [
+                    {
+                        "type": "p",
+                        "children": [
+                            {"text": "this is a "},
+                            {
+                                "children": [
+                                    {"text": ""},
+                                    {
+                                        "type": "a",
+                                        "children": [{"text": "slate link"}],
+                                        "data": {
+                                            "link": {
+                                                "internal": {
+                                                    "internal_link": [
+                                                        {
+                                                            "@id": "%s/image-1"
+                                                            % self.portal.absolute_url(),
+                                                            "title": "Image 1",
+                                                        }
+                                                    ]
+                                                }
+                                            }
+                                        },
+                                    },
+                                    {"text": ""},
+                                ],
+                                "type": "strong",
+                            },
+                            {"text": " inside some text"},
+                        ],
+                    }
+                ],
+            },
+            "6b2be2e6-9857-4bcc-a21a-29c0449e1c68": {"@type": "title"},
+        }
+        res = self.deserialize(blocks=blocks)
+        value = res.blocks["2caef9e6-93ff-4edf-896f-8c16654a9923"]["value"]
+        link = value[0]["children"][1]["children"][1]
+        resolve_link = link["data"]["link"]["internal"]["internal_link"][0]["@id"]
+        self.assertTrue(resolve_link.startswith("../resolveuid/"))
+
+    def test_slate_simple_link_deserializer(self):
```

Comment:
@tiberiuichim can you elaborate what this test checks in comparison to test_slate_internal_link_deserializer. I don't see it at first sight.

Reply:
There's now two ways of doing links with volto-slate: the "old" way, with the sidebar popup and the 3 tabs and the "new" way, which also uses a much simpler data model. The "internal link" is for the "old way", the "simple link" is for the new way.